### PR TITLE
Add xp orb spawn reason for frogs

### PR DESCRIPTION
**Pull request by Lulu13022002**
Track the spawn reason of the xp orb dropped when frogs breeds

### CODE REVIEW

**Review by Owen1212055 — APPROVED** [review]

**Review by Machine-Maker — APPROVED** [review]
I think this patch should mark the ExperienceOrb ctor that doesn't take a SpawnReason as deprecated. Just to increase the likelihood
that usages of it are caught earlier.

**Comment by Lulu13022002** [comment]
I can deprecate the default constructor if you wants ? Also there is a decompile issue in the getBrain method return which needs to be cast into a Brain\<Frog\> before merging this one and the new event

**Comment by Machine-Maker** [comment]
Yeah, add the `@Deprecated` to that constructor.

**Review by kennytv — APPROVED** [review]

File: patches/server/0125-ExperienceOrbs-API-for-Reason-Source-Triggering-play.patch [hunk 1]
```diff
@@ -67,6 +67,7 @@ index b3433ce9c722bdab81848a6c2d121ca510c48509..227aca795efc99c4f81dfb30c00d31d2
 +        nbttagcompound.put("Paper.ExpData", comp);
 +    }
  
++    @Deprecated
```
Comment (Machine-Maker) [inline patches/server/0125-ExperienceOrbs-API-for-Reason-Source-Triggering-play.patch:71]: Now add the `@DoNotUse` annotation to it and remove the `@Deprecated` annotation. That annotation will cause there to be build errors if the annotated method or constructor is used anywhere in the source.

This was just added the other week.

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

File: patches/server/0125-ExperienceOrbs-API-for-Reason-Source-Triggering-play.patch (hunk 1 above)
Comment (Lulu13022002) [reply patches/server/0125-ExperienceOrbs-API-for-Reason-Source-Triggering-play.patch:71]: Fixed i have also cast the brain frog to resolve the build error

**Review by Machine-Maker — APPROVED** [review]

**Review by electronicboy — APPROVED** [review]

**Pull request merged**